src/InvertedIndices.jl

```diff
@@ -160,4 +160,7 @@ end
 @inline Base.to_indices(A, I::Tuple{Not{<:AbstractArray{Bool, 1}}}) = to_indices(A, (eachindex(IndexLinear(), A),), I)
 @inline Base.to_indices(A, I::Tuple{Not{<:Union{Array{Bool}, BitArray}}}) = to_indices(A, (eachindex(A),), I)
 
+@inline Base.getindex(nt::NamedTuple, I::InvertedIndex{Symbol}) = nt[filter(!=(I.skip), keys(nt))]
```

Comment:
following on the discussion from the previous PR opened by @ararslan  - what is the status in Compat.jl? Can these be also safely used pre Julia 1.7 or not?

Reply:
Don't see this namedtuple indexing after a cursory look at Compat.jl. Never used or looked at that package before, so cannot say much more.

Reply:
What I mean by my question is that, if I understood @ararslan correctly, we should first check the status of Compat.jl before merging this PR.
InvertedIndices.jl currently guarantees compatibility with Julia 1.1 or later, so we should make sure all codes using InvertedIndices.jl will work under these different versions of Julia (I understand that the indexing without `Not` was introduced in Julia 1.7, but at least we should decide before merging if we want to accept the fact that this package does not strictly support all Julia versions since 1.1 in the same way).

Reply:
How is the current status of Compat.jl relevant here? InvertedIndices doesn't depend on Compat anyway, and `using InvertedIndices` won't bring the new (1.7) namedtuple indexing.
Whether these `Not` indexing methods are defined on 1.6- versions or not, the result would be effectively the same: a `MethodError`.

Reply:
Also, implementations based on @ararslan's suggestions, 
```
@inline Base.getindex(nt::NamedTuple, I::InvertedIndex{Symbol}) = (; (k => nt[k] for k in keys(nt) if k != I.skip)...)
@inline Base.getindex(nt::NamedTuple, I::InvertedIndex{NTuple{N, Symbol}}) where {N} = (; (k => nt[k] for k in keys(nt) if k ∉ I.skip)...)
```
fail the `@inferred` tests.

Reply:
> How is the current status of Compat.jl relevant here?

We would start depending on Compat.jl.

> using `InvertedIndices` won't bring the new (1.7) namedtuple indexing.

We could discuss if we want to bring it or not.

---

My point is that if the user writes `nt[Not((:a, :b))]` it should work under all supported Julia versions (i.e. anything starting from 1.1) - this is the contract we currently have in InvertedIndices.jl. If we were to change it we need to discuss this.

The reason why I am so strict is that InvertedIndices.jl is used by 609 packages directly or indirectly so we cannot lightly make changes in this package (even if they are "borderline" decisions). We have to have a detailed consideration before we make any change here.

Reply:
It's hard to say something definite for me here, as I wasn't aware of the contract:
> it should work under all supported Julia versions (i.e. anything starting from 1.1) - this is the contract we currently have in InvertedIndices.jl

Depending on how far this contract goes, it could be possible to use the current implementation in 1.7+, and a type-unstable implementation (see above) in earlier versions. Still, some tests would only work on 1.7+: the `@inferred` ones.

Reply:
Having read the docs, I thought that `x[Not(i)]` should effectively give the complement to `x[i]`. This holds with the current implementation: when `x[i]` works, `x[Not(i)]` does as well.

Reply:
That is why I call this case "borderline".

I have also checked the docstring.

If we want to merge this PR (which I personally think we want eventually when we sort out what to do) we should update the docstring to describe the behavior implemented.

Maybe the easiest solution is to document in a separate section of the docstring that this feature is supported only for Julia 1.7.0 and we will be fine). In the new part of the docstring also can you please add that in this case indexing is with names not numbers and that bounds (in this case if the names are present in the queried `NamedTuple`) are not checked.

Reply:
What do you think of the current implementation? All tests pass on 1.6 and 1.7, don't have earlier versions installed.

Reply:
Looks good. @ararslan - OK with you?

But can you please add the description to the docstring as we have a bit different rules here than for normal `Not` anyway. Thank you!